docs/addons/efs-csi-driver.md

```diff
@@ -0,0 +1,55 @@
+# EFS CSI Driver Amazon EKS Add-on
+
+The `EFS CSI Driver Amazon EKS Add-on` provides a CSI interface that allows Kubernetes clusters running on AWS to manage the lifecycle of Amazon EFS volumes for persistent storage.
+EFS CSI driver supports both dynamic and static provisioning of storage
+
+A couple of things to note:
+
+- Driver is not compatible with Windows-based container images
+- The number of replicas to be deployed must be less or equal to the number of nodes in the cluster
+
+For more information on the driver, please review the [user guide](https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html).
+
+## Prerequisites
+
+- The EFS file system itself must be created in AWS separately as the driver uses the EFS for storage, but it does not create it.
```

Comment:
For a typical use case, do you think it makes sense to create the actual FS as well? Would it be a valuable feature? 
If yes, we can add it as a follow-up feature. this looks good.

Reply:
I think this would be useful, but it should be optional in case they have already created the EFS

Reply:
For sure. I am just thinking about an enterprise use case. E.g. they want to deploy an app that is using a file system. This will roll out to dev/test/prod. If we don't supply this feature, will it mean that someone will have to roll it out by accessing those account directly? I am thinking of a workaround that would leverage a custom `spi.ResourceProvider` in the platform to do this with https://docs.aws.amazon.com/cdk/api/v1/docs/aws-efs-readme.html.

Let's create an issue, I am going to get interns and will assign it to them.